include/Makefile.am

```diff
@@ -22,6 +22,7 @@ noinst_HEADERS = \
 	nccl_ofi_msgbuff.h \
 	nccl_ofi_param.h \
 	nccl_ofi_pthread.h \
+	nccl_ofi_kvstore.h \
```

Comment:
What's the upside of doing it this way, instead of getting everything to compile with C++ and then just having a map?  I'm not sure the wrapper buys us anything useful.  Is UTHASH just hopeless to compile under C++?

Reply:
Everything already does successfully compile with C++ as far as I'm aware. 

I've gone into this before but the main concern is that you cannot just malloc some bytes and cast it to a type in C++ and not have problems down the line. Constructors and destructors must run. There's way too much existing code relying on this for it to be safe to introduce new code using new types with this new requirement without some small temporary set of of boundaries.

Reply:
Especially since we're only using two compilers that are known to play nice with C, the situation isn't nearly as dire as you state.  Especially if the structures are all "extern C", those two compilers aren't going to do anything stupid.  So I think you're still over-thinking this too much, but I guess I get it. This isn't a sustainable path, and not an approach we're going to take for any of the other interfaces, but I guess it's worth it to get UTHash out of the way sooner.

Reply:
We talked about this one 1:1, and I can see the value of this wrapper.  I don't want this to be a pattern for future changes, but I see this one.  And dumping UTHash asap has a certain value to it.